### DISCUSSION TITLE
More trouble with Supabase - switch to Firebase for real this time?

### DISCUSSION THREAD

**Discussion by ykdojo** [ykdojo/defaang#242] [Tech]
Over the course of the past few months of experimenting with Supabase, I've found that I've had so much trouble with it.

For example:

1. Setting up auth for frontend & back was not straightforward at all. More details in [this discussion](https://github.com/ykdojo/defaang/discussions/139).

In comparison, there's a NextAuth adapter for Firebase, which seems simple to use.

2. Making a backup and restoring the database from it has been troublesome and I'm still yet to figure out a solution for it: https://github.com/ykdojo/defaang/issues/240

3. There are other small things like lack of documentation, not being to send confirmation emails from custom domains unless we set up our own email server, etc.

What do you think about switching to Firebase - or any other alternative for that matter?

**Comment by epicadidash** [comment]
@ykdojo  pocketbase qualifies all of them except the use of custom domain where firebase is better but Pocketbase is a lot faster than firebase in all its functions and creation of backup is automatic

**Reply by ykdojo** [reply]
Is there a NextAuth adapter for it?

**Reply by epicadidash** [reply]
No there is not . Only Firebase has its adapter built in but you can use prisma .

**Reply by ykdojo** [reply]
How would you build auth with pocketbase?
